projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts

```diff
@@ -42,34 +42,181 @@ let NEXT_ID = 0;
 })
 
 export class IgxButtonGroupComponent implements AfterViewInit {
+    /**
+     * @hidden
+     */
     @ViewChildren(IgxButtonDirective) public buttons: QueryList<IgxButtonGroupComponent>;
 
+    /**
+     * An @Input property that sets the value of the `id` attribute. If not set it will be automatically generated.
+     * ```html
+     *  <igx-buttongroup [id]="'igx-dialog-56'" [multiSelection]="!multi" [values]="alignOptions">
+     * ```
+     */
     @HostBinding('attr.id')
     @Input()
     public id = `igx-buttongroup-${NEXT_ID++}`;
 
+    /**
+     * Allows you to set a style using the `itemContentCssClass` input.
+     * The value should be the CSS class name that will be applied to the button group.
```

Comment:
@alexalex972 Can you check whether the class is applied to the button group or it's applied to the item content. If it's to the item content, please update the doc with the correct description.

Reply:
@mpavlinov I've checked it and it's applied to the igx-button-group.